docs/cow-protocol/reference/contracts/periphery/composable_cow.md

```diff
@@ -0,0 +1,520 @@
+    bytes calldata offchainInput,
+    bytes32[] calldata proof
+) external view returns (GPv2Order.Data memory order, bytes memory signature);
+```
+
+This function will:
+
+1. Determine if `owner` is a `safe`, and provide the `SignatureVerifierMuxer` appropriate formatting for the `ERC-1271` signature submission to CoW Protocol.
+2. If not a `safe`, format the `ERC-1271` signature according to `abi.encode(domainSeparator, staticData, offchainData)`.
+
+Subsequently, `ComposableCoW` will:
+
+1. Check that the order is authorized.
+2. Check that the order type supports discrete order generation (ie. `IConditionalOrderGenerator`) by using `IERC165` (and `revert` if not, allowing the watch-tower to prune invalid monitored conditional orders).
+3. Call `getTradeableOrder` on the handler to get the discrete order ([`GPv2Order.Data`](../core/settlement.md#gpv2orderdata-struct)).
+4. Generate the signing data as above. 
+
+## Indexing
+
+* `ConditionalOrderCreated(address indexed owner, ConditionalOrderParams params)`
+* `MerkleRootSet(address index owner, bytes32 root, Proof proof)`
+
+### Custom error codes
+
+* `ProofNotAuthed()` - the proof is not authorized (merkle root incorrect)
+* `SingleOrderNotAuthed()` - the single order is not authorized
+* `SwapGuardRestricted()` - the swap guard did not pass verification
+* `InvalidHandler()` - the handler is not a valid conditional order
+* `InvalidFallbackHandler()` - the fallback handler is not a valid conditional order
+* `InterfaceNotSupported()` - the handler does not support the `IConditionalOrder` interface
+
+:::tip Keep your orders watched
+
+A conditional order developer **SHOULD** use these error codes to ensure that the conditional order is well-formed and not garbage collected / rate limited by a watch-tower.
+
+:::
+
+* `OrderNotValid(string)` - the `staticInput` parameters are not valid for the conditional order
+* `PollTryNextBlock(string)` - signal to a watch-tower that polling should be attempted again
+* `PollTryAtBlock(uint256 blockNumber, string)` - signal to a watch-tower that polling should be attempted again at a specific block number
+* `PollTryAtEpoch(uint256 timestamp, string)` - signal to a watch-tower that polling should be attempted again at a specific epoch (unix timestamp)
+* `PollNever(string)` - signal to a watch-tower that the conditional order should not be polled again (delete)
+
+## Off-chain
+
+### Watch-tower
+
+As these orders are not automatically indexed by the CoW Protocol, there needs to be some method of relaying them to the Order Book API for inclusion in a batch. 
+
+This is the responsibility of a [watch-tower](https://github.com/cowprotocol/watch-tower). CoW Protocol runs a watch-tower that will monitor the `ConditionalOrderCreated` event, and relay the discrete orders to the Order Book API.
+
+There is also a [DAppNode package for running a watch-tower](https://github.com/cowprotocol/dappnodepackage-cow-watch-tower).
```

Comment:
We already discussed the watchtower quite a bit before introducing it. Maybe we should move this to the architecture section? It's part of why we have conditional orders, so it makes sense to me, even if it isn't strictly contract-related.